node/src/components/deploy_acceptor.rs

```diff
@@ -101,120 +184,586 @@ impl DeployAcceptor {
     ) -> Effects<Event> {
         let verification_start_timestamp = Timestamp::now();
         let mut cloned_deploy = deploy.clone();
-        let mut effects = Effects::new();
-        let is_acceptable = cloned_deploy.is_acceptable(&self.chain_name, &self.deploy_config);
-        if let Err(error) = is_acceptable {
-            // The client has submitted an invalid deploy. Return an error to the RPC component via
-            // the responder.
-            if let Some(responder) = maybe_responder {
-                effects.extend(responder.respond(Err(Error::InvalidDeploy(error))).ignore());
-            }
-            effects.extend(
-                effect_builder
-                    .announce_invalid_deploy(deploy, source)
-                    .ignore(),
-            );
-            return effects;
-        }
-
-        let account_key = deploy.header().account().to_account_hash().into();
-
-        // Verify account if deploy received from client and node is configured to do so.
-        if source.from_client() && self.verify_accounts {
+        let acceptable_result =
+            cloned_deploy.is_config_compliant(&self.chain_name, &self.deploy_config);
```

Comment:
I think `is_config_compliant()` can be called directly on the `deploy`, because the `self` parameter of `is_config_compliant()` doesn't actually need to be mutable.

Therefore, the possibly expensive `clone` in line #186 might not be needed as well.